File: stack_queues/232_implement_queue_using_stacks.py
class MyQueue:

    def __init__(self):
        self.newestStack = []
        self.oldestStack = []

    def push(self, x: int) -> None:
        self.newestStack.append(x)

    def shiftStacks(self):
        if not self.oldestStack: # if oldStack is empty
            while self.newestStack:
                self.oldestStack.append(self.newestStack.pop())

    def pop(self) -> int:
        self.shiftStacks()
        if self.oldestStack:
            return self.oldestStack.pop()
        else:
            return -1

    def peek(self) -> int:
        self.shiftStacks()
        if self.oldestStack:
            return self.oldestStack[-1]
        else:
            return -1
        
    def empty(self) -> bool:
        return not self.newestStack and not self.oldestStack
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: stack_queues/queue/232_implement_queue_using_stacks.java
class MyQueue {

    private Stack<Integer> stack1;
    private Stack<Integer> stack2;
    private int front; 
    
    public MyQueue() {
        stack1 = new Stack<Integer>();
        stack2 = new Stack<Integer>();
        
    }   
    
    public void push(int x) {
        if (stack1.empty()) front = x;
        
        while (!stack1.empty()) {
            stack2.push(stack1.pop());
        }
        stack2.push(x);
        while (!stack2.empty()) {
            stack1.push(stack2.pop());
        }

    }
    
    public int pop() {
        int res = stack1.pop();
        if (!stack1.empty()) {
            front = stack1.peek();    
        }
        return res;
    }
    
    public int peek() {
        return front;
    }
    
    public boolean empty() {
        return stack1.empty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */